stonesoup/types/particle.py

```diff
@@ -25,3 +30,58 @@ def __init__(self, state_vector, weight, parent=None, *args, **kwargs):
     @property
     def ndim(self):
         return self.state_vector.shape[0]
+
+
+class Particles(Type):
+    """
+    Particle type
+
+    A collection of particles. Contains a state and weight for each particle
+    """
+    state_vector: StateVectors = Property(default=None, doc="State vectors of particles")
+    weight: MutableSequence[Probability] = Property(default=None, doc='Weights of particles')
+    parent: 'Particles' = Property(default=None, doc='Parent particles')
+    particle_list: MutableSequence[Particle] = Property(default=None,
+                                                        doc='List of Particle objects')
+
+    def __init__(self, state_vector=None, weight=None, parent=None, particle_list=None,
+                 *args, **kwargs):
+        if state_vector is not None:
+            assert particle_list is None,\
+                "Particles object cannot use both state_vector and particle_list"
+
+        if particle_list and isinstance(particle_list, list):
+            state_vector = StateVectors([particle.state_vector for particle in particle_list])
+            weight = np.array([Probability(particle.weight) for particle in particle_list])
+            parent_list = [particle.parent for particle in particle_list
+                           if particle.parent is not None]
```

Comment:
would we not want to keep `None` where a particle has no parent? Or is a misalignment between parents and children not an issue?

Reply:
Good point. I don't think that situation should ever come up, but the code can definitely be clearer. 

@sdhiscocks  Is there ever a case when some particles would have parents and others wouldn't? My code doesn't like being given None as a particle, so I'll need to handle that if it can happen. If not then I'll change this section of the code to check that either all particles have parents or none do and handle appropriately.

Reply:
I can't think of a scenario, although we don't have any practical use of parent yet anyway.  As you suggest, something like checking the parent list is either empty (all `None`) or correct size (all not `None`) and raising a `ValueError` otherwise is a good approach.